application-analytics-ui/src/main/resources/Analytics/Code/Macros/Filter.xml

```diff
@@ -0,0 +1,501 @@
+      </supportsInlineMode>
+      <visibility>
+        <cache>0</cache>
+        <disabled>0</disabled>
+        <displayType>select</displayType>
+        <freeText>forbidden</freeText>
+        <largeStorage>0</largeStorage>
+        <multiSelect>0</multiSelect>
+        <name>visibility</name>
+        <number>6</number>
+        <prettyName>Macro visibility</prettyName>
+        <relationalStorage>0</relationalStorage>
+        <separator>|</separator>
+        <separators>|</separators>
+        <size>1</size>
+        <unmodifiable>0</unmodifiable>
+        <values>Current User|Current Wiki|Global</values>
+        <classType>com.xpn.xwiki.objects.classes.StaticListClass</classType>
+      </visibility>
+    </class>
+    <property>
+      <async_cached>0</async_cached>
+    </property>
+    <property>
+      <async_context>
+        <value>wiki</value>
+      </async_context>
+    </property>
+    <property>
+      <async_enabled>0</async_enabled>
+    </property>
+    <property>
+      <code>{{velocity}}
+#set ($discard =  $xwiki.ssx.use('Analytics.Code.Macros.Filter'))
+#if ($services.licensing.licensor.hasLicensureForEntity(
+    $services.model.createDocumentReference('', ['Analytics', 'Code'], 'ConfigurationClass')))
+#set ($placeholderStart = '2007-01-01')
+#set ($placeholderEnd = $datetool.get('yyyy-MM-dd'))
+#if($request.analyticsStartDate &amp;&amp; $request.analyticsStartDate != '')
+  #set ($placeholderStart = $request.analyticsStartDate)
+#end
+#if($request.analyticsEndDate &amp;&amp; $request.analyticsEndDate != '')
+  #set ($placeholderEnd = $request.analyticsEndDate)
+#end
+{{html clean="false"}}
+&lt;form id="analyticsFilter" class="form-horizontal"&gt;
+  &lt;div class="row"&gt;
+    &lt;div class="col-sm-2"&gt;
+      &lt;!-- Start Date Label &amp; Input Group --&gt;
+      &lt;div&gt;
+        &lt;label for="analyticsStartDate"&gt;$services.localization.render('rendering.macro.analyticsFilter.startDate')&lt;/label&gt;
+        #set ($dateTimePickerParams = {
+        'id': 'analyticsStartDate',
+        'name': 'analyticsStartDate',
+        'data-format': 'yyyy-MM-dd',
+        'placeholder': $placeholderStart
+        })
+        #dateTimePicker($dateTimePickerParams)
+      &lt;/div&gt;
+    &lt;/div&gt;
+    &lt;div class="col-sm-2"&gt;
+      &lt;!-- End Date Label &amp; Input Group --&gt;
+      &lt;div&gt;
+        &lt;label for="analyticsEndDate"&gt;$services.localization.render('rendering.macro.analyticsFilter.endDate')&lt;/label&gt;
+        #set ($dateTimePickerParams = {
+        'id': 'analyticsEndDate',
+        'name': 'analyticsEndDate',
+        'data-format': 'yyyy-MM-dd',
+        'placeholder': $placeholderEnd
+        })
+        #dateTimePicker($dateTimePickerParams)
+      &lt;/div&gt;
+    &lt;/div&gt;
+    &lt;!-- Submit Button --&gt;
+    &lt;div class="col-sm-2"&gt;
+      &lt;div&gt;
+        &lt;div&gt;
+          &amp;nbsp;
+        &lt;/div&gt;
```

Comment:
This is not ok

Reply:
Oki! I will try to find another way to align the apply button